### PR TITLE
README

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: handlers/discard/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# non-Logger for slog.Logger
+
+The `discard` handler is a placeholder to avoid having to conditionally decide if using a logger
+or not. `discard` will handle Panic() and Fatal() correctly, but everything else will be discarded.
+
+for Panic/Fatal messages, the standard `log.Output()` will be called. fields and callstack are lost.
```
Comment (karasz) [inline handlers/discard/README.md:8]: it would be nice to have here some wording that the "standard" is stdlib.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# slog, a back-end agnostic interface for structured logs
+
+[![Go Reference](https://pkg.go.dev/badge/github.com/darvaza-proxy/slog.svg)](https://pkg.go.dev/github.com/darvaza-proxy/slog)
+
+[`slog.Logger`](#interface) provides a simple standardised interface for structured logs for libraries. It supports [six log levels](#log-levels) and [fields](#fields) with unique string labels (keys).
+
+## Interface
+Every method of this interface, with the exception of [`Print()`](#print), returns a `slog.Logger` so it can be daisy chained when composing a log entry.
+A log entry begins setting the [level](#log-levels) followed by optional addition of [fields](#fields) and a [call stack](#call-stack) and ends with a message calling a [Print](#print) method.
```
Comment (karasz) [inline README.md:9]: A log entry begins with setting the [level](#log-levels) which is followed by ....

File: README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+Based on the specified [level](#log-levels) an entry could be [enabled or disabled](#enabled), and calls to methods on disabled entries will cause no action unless it's used to create a new entry with a [level](#log-levels) that is enabled.
```
Comment (karasz) [inline README.md:10]: an entry can be.. Calls to methods on disabled....

File: README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## Log Levels
+An `slog.Logger` entry can be of one of 6 levels, of which Fatal is expected to end the execution just like the standard `log.Fatal()`right after adding the log entry, and Panic to raise a recoverable panic like `log.Panic()`.
```
Comment (karasz) [inline README.md:13]: A `slog.Logger` entry can have one of 6 levels

File: README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ 3. Warn
+ 4. Error
+ 5. Fatal
+ 6. Panic
+
+New log entries can be created by calling the named shortcut methods (`Debug()`, `Info()`, `Warn()`, `Error()`, `Fatal()`, and `Panic()`) or via `WithLevel(level)`.
+
+
+## Enabled
+A log entry is considered Enabled if the back-end would actually log entries of the specified level.
```
Comment (karasz) [inline README.md:26]: back-end? aren't those handlers?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+When a logger is not `Enabled()` string formatting operations or fields and stack commands are not performed, but when it's useful to know if certain level is *Enabled* before deciding how much detail we want to log. For this purpose `WithEnabled()` can be used like:
```
Comment (karasz) [inline README.md:27]: are not performed, but sometimes is useful to know if a certain level is *Enabled* before deciding how much detail we want to log, for this purpose one can use `WihEnabled()` like this:

File: README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+```go
+if log, ok := logger.Debug().WithEnabled(); ok {
+	log.WithField("request", req).Print("Let's write detailed debug stuff")
+} else if log, ok := logger.Info().WithEnabled(); ok {
+	log.Print("Let's write info stuff instead")
+}
+```
+
+Regardless the _Enabled_ state, Fatal and Panic levels are always expected to exit/panic accordingly.
```
Comment (karasz) [inline README.md:36]: Logs of Fatal and Panic level are expected to exit/panic regardless of the _Enabled_ state.

File: README.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+## Fields
+In `slog` fields are unique key/value pairs where the key is a non-empty script and the value could be anything.
```
Comment (karasz) [inline README.md:39]: In `slog` fields are unique key/value pairs where the key is a non-empty string and the value could be any type.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+## Call Stack
+A Call stack is attached to a log entry considering the given distance to a caller/initiator function.
+
+## Print
+`slog.Logger` support three Print methods mimicking their equivalent on the `fmt` library. `Print()`, `Println()`, and `Printf()` that finally attempt to emit the log entry with the given message and any previously attached [Field](#fields).
```
Comment (karasz) [inline README.md:45]: in the `fmt` package from the standard library

File: README.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+## Standard *log.Logger
+To support the standard `*log.Logger`, `slog` provides a `io.Writer` interface connected to a handler function that is expected to parse the entry and call a provided `slog.Logger` as appropriate. This _writer_ is created by calling `NewLogWriter` with the logger and the handler function and it can then be passed to `log.New()` to create the `*log.Logger`.
```
Comment (karasz) [inline README.md:48]: In order to be compatible with the standard library provided `log.Logger`, `slog` provides an `io.Writer` interface connected to a handler......

by calling `NewLogWriter` and passing the logger and the handler functions.

File: README.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+Alternatively a generic handler is provided when using `NewStdLogger()`.
+
+## Handlers
+
+A handlers is an object that implements the `slog.Logger` interface.
```
Comment (karasz) [inline README.md:54]: A handler is an object that implements the `slog.Logger` interface.

**Review by karasz — CHANGES_REQUESTED** [review]
Some minor edits if you will.

File: handlers/discard/README.md (hunk 1 above)
Comment (amery) [reply handlers/discard/README.md:8]: something like this @karasz ?
```diff
 
-for Panic/Fatal messages, the standard `log.Output()` will be called. fields and callstack are lost.
+for Panic/Fatal messages, the [Go standard logger](https://pkg.go.dev/log) will be called. fields and callstack are lost.
 
 ```

**Comment by amery** [comment]
@karasz amended

File: README.md (hunk 10 above)
Comment (karasz) [reply README.md]: slog provides an interface

File: README.md (hunk 6 above)
Comment (karasz) [reply README.md]: Still makes no sense :(

File: README.md (hunk 8 above)
Comment (karasz) [reply README.md]: I don't think that script is the word you are after, or if it is then I didn't understood the sentence.

**Comment by karasz** [comment]
Starting to feel like a grammar nazi.

File: README.md (hunk 8 above)
Comment (amery) [reply README.md]: it's supposed to be non-empty string obviusly

Comment (amery) [reply README.md]: @karasz I do see _string_ on the latest incarnation of the commit

**Comment by amery** [comment]
new variant of the Enabled paragraph pushed @karasz

File: README.md (hunk 6 above)
Comment (amery) [reply README.md]: newly reworded variant pushed @karasz

**Review by karasz — APPROVED** [review]
LGTM

**Pull request merged**